server/controllers/profile.js

```diff
@@ -38,4 +36,35 @@ exports.loadProfile = asyncHandler(async (req, res, next) => {
       profile: profile,
     },
   });
+});
+
+// @route GET /profile/sitter
+// @desc Get all profile data
+// @access Private
+exports.getAllSitters = asyncHandler(async (req, res) => {
+  const result = [];
+  const profiles = await Profile.find({ isSitter: true, price: { $ne: null } })
+  if (profiles) {
+    profiles.filter((profile) => {
+      const address = profile.address;
+      if (address.toLowerCase().search(req.query.location.toLowerCase()) > -1) {
+        result.push(profile);
+      }
+    })
+    if (result.length) {
```

Comment:
Just return the result without any condition. It's up to the client to decide what to render when it's empty :smiley:

Reply:
done the changes